server/src/GraphQL/Queries/Inspector.php

```diff
@@ -0,0 +1,186 @@
+     * @return mixed
+     */
+
+    public function index($rootValue, array $args, GraphQLContext $context, ResolveInfo $resolveInfo)
+    {
+        return true;
+    }
+
+
+    public function packages($rootValue, array $args, GraphQLContext $context, ResolveInfo $resolveInfo)
+    {
+        $file = __DIR__ . "/../../../composer.json";
+        $file2 = __DIR__ . "/../../../composer.lock";
+        $packages = json_decode(file_get_contents($file2), true)['packages'];
+        $requirePackages = json_decode(file_get_contents($file), true)['require'];
+        $requireDevPackages = json_decode(file_get_contents($file), true)['require-dev'];
+
+        $publishableProviders = App::make('Gernzy\PublishableProviders');
+        $paymentProviderServices = App::make('Gernzy\PaymentProviderServices');
+
+        $paymentProviderInformation = [];
+        foreach ($paymentProviderServices  as $key => $value) {
+            $instance = new $value();
+            array_push($paymentProviderInformation, [
+                'provider_name' => $instance->providerName(),
+                'provider_log' => $instance->logFile(),
+                'provider_class' => $value
+            ]);
+        }
+
+        if (!$eventMapping = config('events')) {
+            throw new GernzyException(
+                'An error occured.',
+                'An error occured when determining the eventMapping. None specified.'
+            );
+        }
+
+        $logFileNames = [];
+        foreach (glob(storage_path() . '/logs/*.log') as $filename) {
+            array_push($logFileNames, basename($filename));
+        }
+
+        $packageDataStructure = [
+            // "packages_lock" => $packages,
+            "require_packages" =>  $requirePackages,
+            "require_dev_packages" =>  $requireDevPackages,
+            "providers" =>  config('app.providers'),
+            "payment_providers" => $paymentProviderInformation,
+            "events" => $eventMapping,
+            "publishable_providers" => $publishableProviders,
+            "laravel_log" => $logFileNames
+        ];
+
+        return json_encode($packageDataStructure);
+    }
+
+    public function logContents($rootValue, array $args, GraphQLContext $context, ResolveInfo $resolveInfo)
+    {
+        $fileToLookFor = $args['filename'];
+        $returnArray = [];
+        foreach (glob(storage_path() . '/logs/*.log') as $filename) {
+            if (basename($filename) == $fileToLookFor) {
+                $file = File::get($filename);
+                $parsed = $this->parseLogFile($file);
+                array_push($returnArray, $fileToLookFor);
+                array_push($returnArray, $parsed);
+            }
+        }
+        return json_encode($returnArray);
+    }
+
+    public function filteredLogContents($rootValue, array $args, GraphQLContext $context, ResolveInfo $resolveInfo)
```

Comment:
@AlexeiaW also a good one to test, or to break out the pure functions and test those (regex etc.)

Reply:
ok, also testing this now with the mock file